xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/Node.java

```diff
@@ -2325,4 +2342,267 @@ public boolean equals(Object obj) {
          */
         publisher;
     }
+
+    /**
+     * A unique identifier for a node, in context of all services in the system.
+     *
+     * The properties that uniquely identify a node are its service, and its nodeId.
+     */
+    public final static class UniqueIdentifier implements Serializable
```

Comment:
Again lombok to the rescue?

Tagging this as `@Data` will create toString, hashCode, equals, methods for you, as well as field getters and setters (when not final).